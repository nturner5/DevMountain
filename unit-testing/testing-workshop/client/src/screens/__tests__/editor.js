import React from 'react'
import {mount} from 'enzyme'
import {Component as Editor} from '../editor'

// default case
test('renders editor form by default', () => {
  const wrapper = render()
  expect(wrapper).toMatchSnapshot()
})

// testing props
test('renders the given title', () => {
  const title = 'The day I dualed Lord Voldemort'
  const wrapper = render({title})
  expect(wrapper.find(sel('title')).node.value).toBe(title)
})

// testing user input
test('adds tag when the user hits enter', () => {
  const newTag = 'interwebs'
  const tagList = ['internet', 'web', 'network']
  const wrapper = render({tagList})
  const tagInput = wrapper.find(sel('tags'))
  changeInputValue(tagInput, newTag)
  keyUpInput(tagInput, 13)
  const tagPills = wrapper.find(sel('tag-pills'))
  expect(tagPills.children()).toHaveLength(4)
  expect(tagPills.find(`[data-tag="${newTag}"]`)).toHaveLength(1)
})

// This component doesn't make any subscriptions to test
// data input and we don't use context directly in this
// component so we have no test for that (and you wont
// often need to) but those are the only two other things
// you need to test with React!

function render(props = {}) {
  const propsToUse = {
    onLoad() {},
    onSubmit() {},
    onUnload() {},
    articleSlug: '',
    title: '',
    description: '',
    body: '',
    tagList: [],
    inProgress: false,
    params: {},
    ...props,
  }
  return mount(<Editor {...propsToUse} />)
}

function changeInputValue(input, value) {
  input.simulate('change', {target: {value}})
}

function keyUpInput(input, keyCode) {
  input.simulate('keyup', {keyCode})
}

function sel(id) {
  return `[data-test="${id}"]`
}

test('I submitted my elaboration and feedback', () => {
  const submitted = true
  expect(true).toBe(submitted)
})

//////// EXTRA CREDIT ////////

// If you get this far, try adding a few more tests,
// then file a pull request to add them as extra credit!
// Learn more here: http://kcd.im/testing-workshop-contributing
