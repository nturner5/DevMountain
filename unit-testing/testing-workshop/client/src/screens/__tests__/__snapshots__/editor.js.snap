exports[`test renders editor form by default 1`] = `
<Editor
  articleSlug=""
  body=""
  description=""
  inProgress={false}
  onLoad={[Function]}
  onSubmit={[Function]}
  onUnload={[Function]}
  params={Object {}}
  tagList={Array []}
  title="">
  <div
    className="editor-page">
    <div
      className="container page">
      <div
        className="row">
        <div
          className="col-md-10 offset-md-1 col-xs-12">
          <ListErrors />
          <form
            onSubmit={[Function]}>
            <fieldset>
              <fieldset
                className="form-group">
                <input
                  className="form-control form-control-lg"
                  data-state-key="title"
                  data-test="title"
                  onChange={[Function]}
                  placeholder="Article Title"
                  type="text"
                  value="" />
              </fieldset>
              <fieldset
                className="form-group">
                <input
                  className="form-control"
                  data-state-key="description"
                  data-test="description"
                  onChange={[Function]}
                  placeholder="What\'s this article about?"
                  type="text"
                  value="" />
              </fieldset>
              <fieldset
                className="form-group">
                <textarea
                  className="form-control"
                  data-state-key="body"
                  data-test="body"
                  onChange={[Function]}
                  placeholder="Write your article (in markdown)"
                  rows="8"
                  value="" />
              </fieldset>
              <fieldset
                className="form-group">
                <input
                  className="form-control"
                  data-test="tags"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Enter tags"
                  type="text"
                  value="" />
                <div
                  className="tag-list"
                  data-test="tag-pills" />
              </fieldset>
              <button
                className="btn btn-lg pull-xs-right btn-primary"
                data-test="submit"
                disabled={false}
                type="submit">
                Publish Article
              </button>
            </fieldset>
          </form>
        </div>
      </div>
    </div>
  </div>
</Editor>
`;
